_pages/extension_sharing_strategies.md

```diff
@@ -0,0 +1,143 @@
+
+## 2. Methods and tools for sharing extensions
+
+### 2.1 [nwb-extensions](https://github.com/nwb-extensions): Extensions GitHub Organization
+
+NDX should be shared via a dedicated [nwb-extensions](https://github.com/nwb-extensions) GitHub organization. The NDX catalog provides the ndx-template and staged-extensions extension template repositories (see below) as well as all public ndx-custom-recipe repositories. In this way, all public extensions are managed in one place and MUST follow the same basic internal organization. This also avoids cluttering up the regular [Neurodata Without Borders GitHub organization](https://github.com/NeurodataWithoutBorders). Finally, this strategy allows for labs/universities/independent-groups to maintain the ndx-template source code repository in their own GitHub organization. The ability to do so, can be important in particular when the developers of the extension are funded by their own grants and/or are applying for funding.
+
+#### 2.1.1 [ndx-template](https://github.com/nwb-extensions/ndx-template): Extension template
+
+The [ndx-template](https://github.com/nwb-extensions/ndx-template) provides a standard Git template repository with a standard setup for creating and sharing extensions and instructions for how to use the template. Using [Cookiecutter](https://cookiecutter.readthedocs.io/en/latest/), users can easily modify the template to create the basic setup for their ndx-template repository. See the [ndx-template](https://github.com/nwb-extensions/ndx-template) repository for details on the organization of template.
+
+#### 2.1.2 [staged-extensions](https://github.com/nwb-extensions/staged-extensions): Creating custom ndx recipes
+
+The [staged-extensions](https://github.com/nwb-extensions/staged-extensions) repository is a place to submit NWB:N extension recipes before they become fully fledged NWB extensions.
+
+#### 2.2.3 ndx-custom-recipe: Custom extension catalog recipe
+
+The NDX Catalog contains a collection of ndx-custom-recipes created by the creators of the extensions. See below for further details.
+
+
+### 2.3 Custom Extension Code Repositories
+
+#### 2.3.1 ndx-custom-recipe: Custom extension catalog recipe
+
+The ndx-custom-recipe is created from the staged-extensions repository as part of the NDX catalog and MUST have the same name as the main namespace of the NDX. This strategy helps ensure that there are no name conflicts between public NDX namespace names.  The ndx-custom-recipy  repository MUST contain the following:
+
+* **/** : root directory of the repo
+   * **/ndx-meta.yaml**: REQUIRED YAML file with the metadata describing the extensions. The YAML file MUST contain a dict with the following keys:
+      * **src: <URL>** : URL to the public repository with the sources of the extension
+      * **pip: <URL>** : URL for installing the extensions from PyPi
```

Comment:
Should we make pip required? What about MATLAB users?

Reply:
I believe, right now for MATLAB we would need to add a simple tool to download the YAML from the ``src`` URL and create the extension code. We could do something similar for PyNWB as well (i.e., without having to have PIP). @rly and @jcfr I believe were working on those details.